네이버 로그인 구현 완료

youngxpepp opened the pull request:
# 네이버 로그인 구현 완료

## 📎 해당 이슈

이슈 링크 (#21 )

## 🛠 구현 내용 

- 네이버 로그인 구현
    - Passport NaverStrategy 추가
    - Application에 initPassport 메서드 추가
    - JwtUtil 추가 및 테스트 작성
    - jsonwebtoken의 async/await wrapper 추가
    - OAuthRouter 추가
- ApplicationFactory 제거
    - 팩토리 패턴을 사용할 만큼 Application을 생성하는 로직이 복잡하지 않다
    - main에서 직접 초기화를 해주는 것이 직관적일 수 있다

## 🙋‍ 리뷰어 참고 사항 
- AbstractService.js -> BaseService.js, 멘토님의 피드백을 반영했습니다.

Park-SooYeon commented on server/src/common/config/passport/NaverStrategy.js:
```diff
@@ -0,0 +1,38 @@
+                    const existedUser = await userRepository.findOne({ socialId: profile?.id });
+
+                    if (existedUser === undefined) {
+                        console.log('existed!!!!!');
```
console.log는 지워주는 것이 좋지 않을까요?👀

youngxpepp replied:
와... 이걸 몰랐네요ㅋㅋ 큰일 날뻔 했습니다
수정해서 다시 올릴게요!

youngxpepp replied:
수정했습니다!

Park-SooYeon reviewed (approved):
멘토님께 리뷰 받은내용 수정하시느라 고생하셨습니다:+1:

dh00n reviewed (approved):
고생하셨습니당

youngxpepp committed:
[FEAT] 네이버 로그인 구현

- Passport NaverStrategy 추가
- Application에 initPassport 메서드 추가
- JwtUtil 추가 및 테스트 작성
- jsonwebtoken의 async/await wrapper 추가
- OAuthRouter 추가

youngxpepp committed:
[FIX] ApplicationFactory 제거

- 팩토리 패턴을 사용할 만큼 Application을 생성하는 로직이 복잡하지 않다
- main에서 직접 초기화를 해주는 것이 직관적일 수 있다
